### PR TITLE
feat: xp nfts contract skeleton

### PR DESCRIPTION
**Pull request by andresaiello**
### Summary
- NFT for XP
- Signature logic to update

## Summary by CodeRabbit

- **New Features**
  - Introduced the `ZetaXP` smart contract for advanced NFT minting, updating, and ownership verification.
  - Added functions for NFT minting and updating with signature verification and event emission.
  - Implemented a structured approach for handling NFT updates and ownership management.

- **Tests**
  - Implemented comprehensive test suite for the ZetaXP contract, covering minting, updates, event emission, and ownership validation.

- **Chores**
  - Updated dependencies to include upgradeable contracts and Hardhat upgrade plugins.
  - Added deployment script for the ZetaXP NFT contract within the ZetaChain ecosystem.

### CODE REVIEW

File: packages/zevm-app-contracts/contracts/xp-nft/xpNFT.sol [hunk 1]
```diff
@@ -0,0 +1,180 @@
...
+        uint256 xpTotal;
+        uint256 level;
+        uint256 testnetCampaignParticipant;
+        uint256 enrollDate;
+        uint256 mintDate;
+        uint256 generation;
+    }
+
+    mapping(uint256 => Data) public data;
+    mapping(uint256 => mapping(uint256 => Task)) public tasks;
```
Comment (lucas-janon) [inline packages/zevm-app-contracts/contracts/xp-nft/xpNFT.sol:30]: I'd rename this one so the code is more understandable:
```suggestion
    mapping(uint256 => mapping(uint256 => Task)) public tasksByTokenId;
```

Comment (lucas-janon) [reply packages/zevm-app-contracts/contracts/xp-nft/xpNFT.sol:30]: At first, I thought it was a global tasks list.

File: packages/zevm-app-contracts/contracts/xp-nft/xpNFT.sol [hunk 2]
```diff
@@ -0,0 +1,180 @@
...
+    struct Data {
+        uint256 xpTotal;
+        uint256 level;
+        uint256 testnetCampaignParticipant;
+        uint256 enrollDate;
+        uint256 mintDate;
+        uint256 generation;
+    }
+
+    mapping(uint256 => Data) public data;
```
Comment (lucas-janon) [inline packages/zevm-app-contracts/contracts/xp-nft/xpNFT.sol:29]:
```suggestion
    mapping(uint256 => Data) public tokenData;
```

File: packages/zevm-app-contracts/test/xp-nft/xp-nft.ts [hunk 3]
```diff
@@ -0,0 +1,152 @@
...
+      to: invitee.address,
+      tokenId: 1,
+    };
+  });
+
+  describe("True", () => {
+    it("Should be true", async () => {
+      expect(true).to.equal(true);
+    });
+  });
```
Comment (lucas-janon) [inline packages/zevm-app-contracts/test/xp-nft/xp-nft.ts:50]:
```suggestion
```

File: packages/zevm-app-contracts/test/xp-nft/xp-nft.ts [hunk 4]
```diff
@@ -0,0 +1,240 @@
...
+    const nftParams: UpdateParam = {
+      ...sampleNFT,
+      sigTimestamp,
+      signature,
+    } as UpdateParam;
+    const tx = zetaXP.mintNFT(nftParams);
+    await expect(tx).to.emit(zetaXP, "NewNFTMinted").withArgs(user.address, 1);
+  });
+
+  it("Should revert if signature it's not correct", async () => {
```
Comment (lucas-janon) [inline packages/zevm-app-contracts/test/xp-nft/xp-nft.ts:71]:
```suggestion
  it("Should revert if signature is not correct", async () => {
```

**Review by fadeev — APPROVED** [review]
utACK

**Pull request merged**